R/get_dataset.R

```diff
@@ -0,0 +1,92 @@
+#' Query a data set using GET and a query URL
+#'
+#' @description
+#' This function provides a method for generating and sending a URL based data query to the
+#' EES API. As a minimum, it requires the dataset_id and indicators flags to be provided.
+#'
+#' Note that the GET command is very limited on the level of logic it can process. For example
+#' there is no way of using GET to make a query that combines different filters with AND logic.
+#' So if you give GET a set of filter items to search on, it can only return all rows containing
+#' any of those items.
+#'
+#' @inheritParams api_url
+#' @param parse Logical flag to activate parsing of the results. Default: TRUE
+#'
+#' @return Data frame containing query results of an API data set
+#' @export
+#'
+#' @examples
+#' get_dataset(
```

Comment:
Long term it'd be good to setup a test-get_dataset.R script in the tests folder that checks this against the `read.csv()` version (that we deployed this morning) that pulls the full data set in, would be a good way to check the paging etc works as intended and gives the full data set back?

Can use the same example_id in both

![image](https://github.com/user-attachments/assets/d633ac8f-c289-4ebf-bbff-fab72b948348)

Not something we can right now though as the read.csv code gets blocked by the basic auth we have on the dev environment

Guess maybe one when we get the /CSV API endpoint, and then we could use that instead?

Reply:
Raised https://github.com/dfe-analytical-services/eesyapi/issues/27